### PR TITLE
DE: 2922 nulls and const

### LINKED ISSUE #23: Validator expects const strings to have type "string","null"
While validating the PerkinElmer-Labchip IDS, this behavior was demonstrated
```
root.properties.systems.items.properties.vendor:
   * 'type' must be ['string', 'null']
root.properties.systems.items.properties.model:
   * 'type' must be ['string', 'null']
root.properties.systems.items.properties.type:
   * 'type' must be ['string', 'null']
```
when the schema lists:
```
          "vendor": {
            "type": "string",
            "const": "PerkinElmer"
          },
          "model": {
            "type": "string",
            "const": "LabChip GXII Touch"
          },
          "type": {
            "type": "string",
            "const": "Protein Characterization System"
          }
```

It would make sense if in this case the validator accepted that the `const`s here only needs to be type `"string"`

### PR DESCRIPTION
**Pull request by setiat**
Fixes #23

### CODE REVIEW

File: __tests__/unit/generic/files/type_check/valid_type_with_const.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "type": "boolean",
+    "const": false
```
Comment (mlipschultz) [inline __tests__/unit/generic/files/type_check/valid_type_with_const.json:3]: Why do you need `"const": false` here?  Wouldn't IDS designers generally leave `const` out if it's not constant?

Comment (setiat) [reply __tests__/unit/generic/files/type_check/valid_type_with_const.json]: updated example.

File: ids_validator/checks/rules_checker.py [hunk 2]
```diff
@@ -47,6 +47,18 @@ def enforce_checks(cls, node: Node, checks: dict):
...
+        # don't allow "null" if "const" is defined
+        # schema will be enforced only when const is not
+        # defined
+        if "const" in node.data:
+            is_valid, msg =  node._check_const_type()
+            if not is_valid:
+                logs += [(msg, Log.CRITICAL.value)]
+                return logs
+            else:
+                return logs
```
[the remark this answers is not in the document]
Comment (setiat) [reply ids_validator/checks/rules_checker.py]: fixed.

**Review by mlipschultz — APPROVED** [review]
Thanks for fixing the example.  Now that I'm looking at your fix, I realize the original was fine.  For some reason, I didn't realize that a boolean type could have a constant `false`.  Sorry!

**Pull request merged**